client/blocks/plan-purchase-features/index.jsx

```diff
@@ -265,24 +253,74 @@ class PlanPurchaseFeatures extends Component {
 		}
 	}
 
+	getPlanHeaderWording() {
+		const { plan, translate } = this.props;
+
+		const planConstObj = plansList[ plan ],
+			title = translate( 'Your site is on a %(planName)s plan', {
+				args: {
+					planName: planConstObj.getTitle()
+				}
+			} );
+
+		let tagLine = translate( 'Unlock the full potential of your site with all the features included in your plan.' );
+
+		if ( plan === PLAN_BUSINESS ) {
+			tagLine = translate( 'Learn more about everything included with Business and take advantage of' +
+				' its professional features.' );
+		}
+
+		return {
+			title: title,
+			tagLine: tagLine
+		};
+	}
+
+	getPlanPurchaseHeader() {
+		const {
+			page,
+			selectedSite,
+			isPlaceholder
+		} = this.props;
+
+		if ( page === 'current-plan' ) {
+			const { title, tagLine } = this.getPlanHeaderWording();
+
+			return [
+				<CurrentPlanHeader
+					selectedSite={ selectedSite }
+					key="currentPlanHeaderFeature"
+					title={ title }
+					tagLine={ tagLine }
+					isPlaceholder={ isPlaceholder }
+				/>,
+				<HappinessSupport
+					selectedSite={ selectedSite }
+					key="hapinessSupportFeature"
+					isPlaceholder={ isPlaceholder }
+				/>
+			];
+		}
+	}
+
 	render() {
+		const { isPlaceholder } = this.props;
+
+		const headerClasses = classNames( 'plan-purchase-features__header', {
+			'is-placeholder': isPlaceholder
+		} );
+
 		return (
 			<div className="plan-purchase-features">
-				{ this.getPlanPurchaseFeatures() }
+				<div className={ headerClasses }>
+					{ this.getPlanPurchaseHeader() }
+				</div>
```

Comment:
If you don't like the monolithic structure of `plan-purchase-features`, I could do as you said and allow for header to be set as a child of the component. However, that just moves a few bit of code (ie `getPlanPurchaseHeader` method) to `current-plan/index`. What are the benefits? Maybe that we wouldn't have to add a new `page` if we were to use `plan-purchase-features` on some other place (and instead just passing different header?). Dunno if that'll be so flexible.